modules/partition-replicator/src/main/java/org/apache/ignite/internal/partition/replicator/handlers/WriteIntentSwitchRequestHandler.java

```diff
@@ -117,12 +117,8 @@ public CompletableFuture<ReplicaResult> handle(WriteIntentSwitchReplicaRequest r
                 .map(tableId -> invokeTableWriteIntentSwitchReplicaRequest(tableId, request, clockService.current(), senderId))
                 .collect(toList());
 
-        // We choose current() to try to avoid compaction of the chosen version (and we make sure it's not below commitTs [if it's a commit]
-        // or txBeginTs [if it's an abort]). But there seems to be no guarantee that the compactor will not remove this version.
-        // TODO: IGNITE-24574 Introduce a mechanism to save the chosen catalog version from being compacted too early.
```

Comment:
This is not what you've implemented. Why is that the case?

Reply:
The comment was suggested to introduce a mechanism that would allow to hold catalog compaction, but it turned out that we can solve the problem at hand a lot easier. `reliableCatalogVersion` is used to make sure the follower has *at least* the version we used when producing the command, so it might exceed the 'exact' value. So, if the 'exact' value was already compacted away, it's not a problem, we can take any value that exceeds it (this is what I do by taking the earliest version of the Catalog currently available).

I added a comment to the Jira issue explaining the change of plans.